sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/authoring/_patch.py

```diff
@@ -0,0 +1,86 @@
+
+# This file is used for handwritten extensions to the generated code. Example:
+# https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/customize_code/how-to-patch-sdk-code.md
+
+from typing import List, Union, Any
+from azure.core.credentials import AzureKeyCredential, TokenCredential
+from azure.core.pipeline.policies import AzureKeyCredentialPolicy
+from ._client import TextAuthoringClient as GeneratedTextAuthoringClient
+
+
+def _authentication_policy(credential):
+    authentication_policy = None
+    if credential is None:
+        raise ValueError("Parameter 'credential' must not be None.")
+    if isinstance(credential, AzureKeyCredential):
+        authentication_policy = AzureKeyCredentialPolicy(name="Ocp-Apim-Subscription-Key", credential=credential)
+    elif credential is not None and not hasattr(credential, "get_token"):
+        raise TypeError(
+            "Unsupported credential: {}. Use an instance of AzureKeyCredential "
+            "or a token credential from azure.identity".format(type(credential))
+        )
+    return authentication_policy
+
+
+class TextAuthoringClient(GeneratedTextAuthoringClient):  # pylint: disable=client-accepts-api-version-keyword
+    """The language service API is a suite of natural language processing (NLP) skills built with
+    best-in-class Microsoft machine learning algorithms. The API can be used to analyze
+    unstructured text for tasks such as sentiment analysis, key phrase extraction, language
+    detection and question answering. Further documentation can be found in :code:`<a
+    href="https://docs.microsoft.com/en-us/azure/cognitive-services/language-service/overview">https://docs.microsoft.com/en-us/azure/cognitive-services/language-service/overview</a>`.
+
+    :ivar text_analysis_authoring: TextAnalysisAuthoringOperations operations
+    :vartype text_analysis_authoring:
+     azure.ai.language.text.authoring.operations.TextAnalysisAuthoringOperations
+    :param endpoint: Supported Cognitive Services endpoint (e.g.,
+     https://:code:`<resource-name>`.cognitiveservices.azure.com). Required.
+    :type endpoint: str
+    :param credential: Credential needed for the client to connect to Azure. Required.
+    :type credential: ~azure.core.credentials.TokenCredential
+    :keyword api_version: Api Version. Default value is "2022-05-01". Note that overriding
+     this default value may result in unsupported behavior.
+    :paramtype api_version: str
+    :keyword int polling_interval: Default waiting time between two polls for LRO operations if no
+     Retry-After header is present.
+    """
+
+    def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, TokenCredential], **kwargs: Any) -> None:
+        super().__init__(
+            endpoint=endpoint,
+            credential=credential,  # type: ignore
+            authentication_policy=kwargs.pop("authentication_policy", _authentication_policy(credential)),
```

Comment:
I noticed when we removed version from the swagger README, it generated it as "unknown" for the authoring client (see [here](https://github.com/Azure/azure-sdk-for-python/blob/d7b27bdd6ea54a745d5bac5ea6853fb152f9c0aa/sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/authoring/_configuration.py#L18)). I was hoping it would be smart enough to use the runtime version, but looks like we'll need to help it a little. :)

We can pass `sdk_moniker` into the client here to correctly set the User-Agent header (which uses the version) and override the "unknown" behavior. This should look something like this:

```
from ..user_agent import USER_AGENT

# .. code ...

def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, TokenCredential], **kwargs: Any) -> None:
    super().__init__(
        endpoint=endpoint,
        credential=credential,  # type: ignore
        authentication_policy=kwargs.pop("authentication_policy", _authentication_policy(credential)),
        sdk_moniker=USER_AGENT,
        **kwargs
)
```